File: backend/core-services/product-service/src/main/java/com/movieticket/product/entity/Product.java
package com.movieticket.product.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "products")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {
    @Id
    private String id;

    private String name;
    private double unitPrice;

    private String pictureUrl;

    @Column(columnDefinition = "TEXT")
    private String description;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private boolean status;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/web/src/types/product.ts
export const ProductType = {
  SINGLE: "SINGLE",
  COMBO: "COMBO",
} as const;

export type ProductType = (typeof ProductType)[keyof typeof ProductType];

export interface Product {
  id: string;
  name: string;
  unitPrice: number;
  pictureUrl: string;
  description: string;
  type: ProductType;
  createdAt: string;
  updatedAt: string;
  status: boolean;
}

export interface ProductFilterParams {
  id?: string;
  name?: string;
  minPrice?: number;
  maxPrice?: number;
  type?: ProductType;
  status?: boolean;
  page?: number;
  size?: number;
}
